models/fates/inst/template.job

```diff
@@ -13,101 +16,105 @@ mkdir -p "@OUTDIR@"
-  mkdir input
-  echo "PEcAn.FATES::recurse.create('input','@DEFAULT@')" | R --vanilla
-
-  cd "@CASEDIR@"
-
-  ## THINGS THAT ARE IN REFCASE
-  # Modifying : env_mach_pes.xml
-  echo "*** Modify XMLs ***"
-./xmlchange -file env_mach_pes.xml -id NTASKS_ATM -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_LND -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_ICE -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_OCN -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_CPL -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_GLC -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_ROF -val 1
-./xmlchange -file env_mach_pes.xml -id NTASKS_WAV -val 1
-./xmlchange -file env_mach_pes.xml -id MAX_TASKS_PER_NODE -val 1
-./xmlchange -file env_mach_pes.xml -id TOTALPES -val 1
-  # Modifying : env_build.xml
-./xmlchange -file env_build.xml -id GMAKE -val make
-./xmlchange -file env_build.xml -id DEBUG -val FALSE
-  # Modifying : env_run.xml
-./xmlchange -file env_run.xml -id REST_N -val 1
-./xmlchange -file env_run.xml -id REST_OPTION -val nyears
-./xmlchange -file env_run.xml -id DATM_CLMNCEP_YR_START -val 1974
-./xmlchange -file env_run.xml -id DATM_CLMNCEP_YR_END -val 2004
-./xmlchange -file env_run.xml -id DIN_LOC_ROOT -val @INDIR@
-./xmlchange -file env_run.xml -id DIN_LOC_ROOT_CLMFORC -val '@INDIR@'
-./xmlchange -file env_run.xml -id DOUT_S -val TRUE
-./xmlchange -file env_run.xml -id DOUT_S_ROOT -val '@CASEDIR@/run'
-./xmlchange -file env_run.xml -id PIO_DEBUG_LEVEL -val 0
-./xmlchange -file env_run.xml -id ATM_DOMAIN_FILE -val 'domain.lnd.@SITE_NAME@.nc'
-./xmlchange -file env_run.xml -id ATM_DOMAIN_PATH -val '@INDIR@/share/domains/domain.clm/'
-./xmlchange -file env_run.xml -id LND_DOMAIN_FILE -val 'domain.lnd.@SITE_NAME@.nc'
-./xmlchange -file env_run.xml -id LND_DOMAIN_PATH -val '@INDIR@/share/domains/domain.clm/'
-./xmlchange -file env_run.xml -id CLM_USRDAT_NAME -val '@SITE_NAME@'
+## RECURSIVELY COPY/SYMLINK REFERENCE INPUTS DIRECTORY (DIN_LOC_ROOT)
+## create folders and symbolic links. Links will later be deleted when non-default files are specified
+mkdir input
+echo "PEcAn.FATES::recurse.create('input','@DEFAULT@')" | R --vanilla
+
+cd "@CASEDIR@"
+
+## THINGS THAT ARE IN REFCASE
+# Modifying : env_mach_pes.xml
+echo "*** Modify XMLs ***"
+#./xmlchange --file env_mach_pes.xml --id NTASKS_ATM --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_LND --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_ICE --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_OCN --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_CPL --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_GLC --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_ROF --val 1
+#./xmlchange --file env_mach_pes.xml --id NTASKS_WAV --val 1
+#./xmlchange --file env_mach_pes.xml --id MAX_TASKS_PER_NODE -val 1
+#./xmlchange --file env_mach_pes.xml --id TOTALPES --val 1
```

Comment:
Commented values should be removed. They will live on forever in the Git history.

Reply:
Unless they are for upcoming features that haven't been implemented yet, in which case you can leave them.

Reply:
More so, I dont know yet if these need to be explicit or not on all machines.  The tricky part is that depending on the machine-specific configs, these may need to be set but on Modex I have these set automatically.  So to be honest I dont know yet if we need these set or not until someone else tries running on a different machine

Reply:
Fair enough.

In general, I don't like that so much of this is hard-coded within the `template.job` file. Given that all of these commands follow a very regular pattern, I think it's a much better idea to do this all from within R so that the user can set any or all of these tags from within the `pecan.xml` file. For instance, something like this:

```r
xmlchange <- function(file, id, value) {
  system2("./xmlchange", c("--file", file, "--id", id, "--value", value))
}

set_xml_values <- function(file, tag_list) {
  for (i in seq_len(tag_list)) {
    xmlchange(file, names(tag_list)[i], tag_list[[i]])
  }

write_fates_xml <- function(casedir, site_name, xml_file = "env_build.xml" ...) {
  dots <- list(...)
  default_tags <- list(NTASKS_ROF = 1, NTASKS_WAV = 1, MAX_TASKS_PER_NODE = 1, TOTALPES = 1, <etc>)
  new_tags <- modifyList(list(CIME_OUTPUT_ROOT = casedir, EXEROOT = file.path(casedir, bld)), dots)
  use_tags <- modifyList(default_tags, new_tags)
  set_xml_vaues(xml_file, use_tags)

settings$run$other_settings <- list(NTASKS_ROF = 1, NTASKS_WAV = 1)
do.call(write_fates_xml, c(list(casedir = settings$run$casedir, site_name = settings$site_name), settings$run$other_settings))
```

...but that's probably best saved for another time.

Reply:
Actually, that is a good plan for a targeted PR that would benefit both CLM and FATES.  I am going to take this nugget and create an issue focused on this sort of "upgrade" to the code